### PR TITLE
Making kpcovr test deterministic by using  svd_solver

### PR DESCRIPTION
**Pull request by rosecers**
In response to #100, removing non-deterministic behavior from the kpcovr_test

### CODE REVIEW

File: tests/test_kpcovr.py [hunk 1]
```diff
@@ -25,7 +25,9 @@ def __init__(self, *args, **kwargs):
         self.X = SFS().fit_transform(self.X)
         self.Y = SFS(column_wise=True).fit_transform(self.Y)
 
-        self.model = lambda mixing=0.5, **kwargs: KPCovR(mixing, alpha=1e-8, **kwargs)
+        self.model = lambda mixing=0.5, **kwargs: KPCovR(
+            mixing, alpha=1e-8, svd_solver=kwargs.pop("svd_solver", "full"), **kwargs
+        )
```
Comment (hurricane642) [inline tests/test_kpcovr.py:30]: Is this really enough to fix the problem? I'm just afraid that there are difficulties because of the way `self.Y` is created.

**Comment by rosecers** [comment]
What issues do you see with `self.Y`?

**Comment by hurricane642** [comment]
> What issues do you see with `self.Y`?
> […](#)

        ```
self.Y = np.array(
            [self.Y, self.X @ np.random.randint(-2, 2, (self.X.shape[-1],))]
        ).T
```

**Comment by rosecers** [comment]
> > What issues do you see with `self.Y`?
> > […](#)
> 
> self.Y = np.array(
> [self.Y, self.X @ np.random.randint(-2, 2, (self.X.shape[-1],))]
> ).T
> 
> The randomness of the data

I totally missed this! Taking another look over

**Comment by rosecers** [comment]
> > > What issues do you see with `self.Y`?
> > > […](#)
> > 
> > 
> > self.Y = np.array(
> > [self.Y, self.X @ np.random.randint(-2, 2, (self.X.shape[-1],))]
> > ).T
> > The randomness of the data
> 
> I totally missed this! Taking another look over

That shouldn't cause an issue, as both KPCovR and PCovR are operating on the same dataset each time, and so randomness in the DS won't cause randomness in the results.

**Comment by hurricane642** [comment]
> > > > What issues do you see with `self.Y`?
> > > > […](#)
> > > 
> > > 
> > > self.Y = np.array(
> > > [self.Y, self.X @ np.random.randint(-2, 2, (self.X.shape[-1],))]
> > > ).T
> > > The randomness of the data
> > 
> > 
> > I totally missed this! Taking another look over
> 
> That shouldn't cause an issue, as both KPCovR and PCovR are operating on the same dataset each time, and so randomness in the DS won't cause randomness in the results.

That's the thing, the results are really the same. There is a difference in the third digit, and it seems to me that this is not so much an error of the algorithm, as the difference due to rounding (that is due to different values there in 4-5 digits after the decimal point).

**Comment by hurricane642** [comment]
> What do you mean? We are addressing the source of the difference, not the difference itself. the switching of SVD solver should fix this.

That's what I'm talking about. I mean, we have two answers, 0.751 and 0.75. Should this be considered the same response (according to the idea of the algorithm)? Or if for the same dataset two algorithms give a difference of 0.001, then one of them doesn't work? I just don't know what the accuracy is here in general.

**Comment by rosecers** [comment]
> > What do you mean? We are addressing the source of the difference, not the difference itself. the switching of SVD solver should fix this.
> 
> That's what I'm talking about. I mean, we have two answers, 0.751 and 0.75. Should this be considered the same response (according to the idea of the algorithm)? Or if for the same dataset two algorithms give a difference of 0.001, then one of them doesn't work? I just don't know what the accuracy is here in general.

It should not be considered the same response, as we discussed in dev

**Comment by hurricane642** [comment]
> > > What do you mean? We are addressing the source of the difference, not the difference itself. the switching of SVD solver should fix this.
> > 
> > 
> > That's what I'm talking about. I mean, we have two answers, 0.751 and 0.75. Should this be considered the same response (according to the idea of the algorithm)? Or if for the same dataset two algorithms give a difference of 0.001, then one of them doesn't work? I just don't know what the accuracy is here in general.
> 
> It should not be considered the same response, as we discussed in dev

Ok, that's it, then no questions asked.

**Review by hurricane642 — APPROVED** [review]

**Comment by ceriottm** [comment]
Randomness in the dataset CAN cause randomness in the result - different
datasets might be harder or easier to fit.
Also why randint? I can see a risk with randint getting a vector where all
the Y are the same, no?
